clusters/build-clusters/01_cluster/Makefile

```diff
@@ -0,0 +1,57 @@
+release_image := registry.svc.ci.openshift.org/ocp/release:4.2.0-0.nightly-2019-10-08-232417
+
+.PHONY: all
+all:
+	#TODO (hongkliu)
+	@echo "target 'all' has not yet implemented"
+
+.PHONY: install-dptp-managed-cluster
+install-dptp-managed-cluster:
+	@echo "installing dptp-managed cluster ..."
+	./install_cluster.sh $(release_image)
+
+###pip install --user yq
+secret_file := "/tmp/secret.yaml"
+client_id := $(shell yq -r .github.client_id $(secret_file))
+client_secret := $(shell yq -r .github.client_secret $(secret_file))
+
+.PHONY: install-configure-github-oauth
+install-configure-github-oauth:
+	@echo "configuring github oauth ..."
+	@echo "replacing the clientSecret in github-secret_secret.yaml ..."
+	@sed -e "s|{{ clientSecret }}|$(client_secret)|g" ./assets/github-secret_secret.yaml > /tmp/github-secret_secret.yaml
+	### the namespace is defined in the yaml
+	oc apply -f /tmp/github-secret_secret.yaml
+	sha256sum /tmp/github-secret_secret.yaml
+	rm -fv /tmp/github-secret_secret.yaml
+	@echo "replacing the clientID in github_oauth.yaml ..."
+	@sed -e "s|{{ clientID }}|$(client_id)|g" ./assets/github_oauth.yaml > /tmp/github_oauth.yaml
+	oc apply -f /tmp/github_oauth.yaml
+	sha256sum /tmp/github_oauth.yaml
+	rm -fv /tmp/github_oauth.yaml
+
+.PHONY: remove-configure-github-oauth
+remove-configure-github-oauth:
+	@echo "removing github oauth ..."
+	oc delete -f ./assets/github-secret_secret.yaml --ignore-not-found=true
+	oc delete -f ./assets/github_oauth.yaml --ignore-not-found=true
+
+.PHONY: set-up-ci-admins
+set-up-ci-admins:
+	@echo "setting up ci-admins"
+	#oc adm policy add-cluster-role-to-group cluster-admin ci-admins
+	oc create -f ./assets/roles.yaml
+
+.PHONY: set-up-auto-scaler
+set-up-auto-scaler:
+	@echo "setting up auto-scaler ..."
+	oc apply -f ./assets/default_clusterautoscaler.yaml
+	@declare -a arr=( $$(oc get MachineSet -n openshift-machine-api -o yaml | yq -r .items[].metadata.name)); for machineset_name in "$${arr[@]}"; \
+	do \
+	  echo "set up machineautoscaler for $${machineset_name}"; \
+	  echo "replacing the machineset_name in machineautoscaler.yaml ..."; \
+	  sed -e "s|{{ machineset_name }}|$${machineset_name}|g" ./assets/machineautoscaler.yaml > "/tmp/machineautoscaler-$${machineset_name}.yaml"; \
+	  oc apply -f "/tmp/machineautoscaler-$${machineset_name}.yaml"; \
```

Comment:
@petr-muller 
I have a loop here of applying the some params from `machineSet` of the cluster to the same yaml file.
In my opinion, it is kind hard for `applyConfig` to have all the specific logics that `Makefile` can do.

Maybe I should not have cared much about automation ... once the cluster is installed, those params are known.
I was trying to avoid the copy/paste by the `loop`.
What do you think about this case?

Reply:
I think these can be solved with the two-step process: Have a helper `Makefile` target that generates the files, but the generated files are then committed to the repo (and therefore applied by applyconfig).